src/utils.c

```diff
@@ -234,3 +234,44 @@ avifBool avifFractionSub(avifFraction a, avifFraction b, avifFraction * result)
     avifFractionSimplify(result);
     return AVIF_TRUE;
 }
+
+avifBool avifToUnsignedFraction(double v, uint32_t * numerator, uint32_t * denominator)
+{
+    if (v < 0 || v > UINT32_MAX) {
+        return AVIF_FALSE;
+    }
+
+    // Good enough tolerance for stopping early.
+    const double relative_tolerance = 1e-10;
+    // Maximum denominator: makes sure that both the numerator and denominator are <= UINT32_MAX.
+    const uint64_t max_d = (v <= 1) ? UINT32_MAX : (uint64_t)floor(UINT32_MAX / fabs(v));
+
+    // Find the best approximation of v as a fraction using continued fractions, see
+    // https://en.wikipedia.org/wiki/Continued_fraction
+    *denominator = 1;
+    uint32_t previous_d = 0;
+    double current_v = v - floor(v);
+    int iter = 0;
+    // Set a maximum number of iterations to be safe but it should converge in < 10 iterations.
```

Comment:
The slowest converging number is golden ratio (1.618...) which requires 25 terms to reach 1e-10 tolerance and 38 terms to reach `double` precision, so we can make a more reasonable guess here, or even assert when >38 iterations happened. And it would be good to test against golden ratio as well.

Reply:
Thanks for the tip. Done.